# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpSender.cpp
# Commit: 99ea4f221834
# Full Hash: 99ea4f221834f706be3738a92183287b54d8f718
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-22 17:00:32
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Make RTCRtpSender::GetStats get ICE stats. r=ng
#   
#   Also fix bugs where we would get duplicate ICE stats. This mostly mattered to
#   WebrtcGlobalInformation since RTCStatsReportInternal does not automatically
#   remove duplicate ids like RTCStatsReport (RTCStatsReport is maplike).
# ==============================================================================

diff -r 5801060dd3d0 -r 99ea4f221834 dom/media/webrtc/jsapi/RTCRtpSender.cpp
--- a/dom/media/webrtc/jsapi/RTCRtpSender.cpp	Thu Dec 22 02:33:56 2022 +0000
+++ b/dom/media/webrtc/jsapi/RTCRtpSender.cpp	Thu Dec 22 02:33:57 2022 +0000
@@ -56,6 +56,7 @@
     : mWindow(aWindow),
       mPc(aPc),
       mSenderTrack(aTrack),
+      mTransportHandler(aTransportHandler),
       mTransceiver(aTransceiver),
       INIT_CANONICAL(mSsrcs, Ssrcs()),
       INIT_CANONICAL(mVideoRtxSsrcs, Ssrcs()),
@@ -146,7 +147,8 @@
   return promise.forget();
 }
 
-nsTArray<RefPtr<dom::RTCStatsPromise>> RTCRtpSender::GetStatsInternal() {
+nsTArray<RefPtr<dom::RTCStatsPromise>> RTCRtpSender::GetStatsInternal(
+    bool aSkipIceStats) {
   MOZ_ASSERT(NS_IsMainThread());
   nsTArray<RefPtr<RTCStatsPromise>> promises(2);
   if (!mSenderTrack || !mPipeline) {
@@ -425,6 +427,12 @@
         return RTCStatsPromise::CreateAndResolve(std::move(report), __func__);
       }));
 
+  if (!aSkipIceStats && GetJsepTransceiver().mTransport.mComponents) {
+    promises.AppendElement(mTransportHandler->GetIceStats(
+        GetJsepTransceiver().mTransport.mTransportId,
+        mPipeline->GetTimestampMaker().GetNow()));
+  }
+
   return promises;
 }
 